packages/dds/sequence/src/revertibles.ts

```diff
@@ -387,11 +387,22 @@ function getSlidePosition(
 	const slide = collection.getSlideToSegment(lref);
 	return slide?.segment !== undefined &&
 		slide.offset !== undefined &&
-		string.getPosition(slide.segment) !== -1
+		string.getPosition(slide.segment) !== -1 &&
+		(pos < 0 || pos > string.getLength())
 		? string.getPosition(slide.segment) + slide.offset
 		: pos;
 }
 
+function isValidRange(start: number, end: number, string: SharedString) {
+	return (
+		start >= 0 &&
+		start <= string.getLength() - 1 &&
```

Comment:
This is a little easier to read. Same for end below.
```suggestion
		start < string.getLength() &&
```